Apromore-Core-Components/Apromore-Portal/src/main/java/org/apromore/portal/dialogController/TokenHandoffController.java

```diff
@@ -0,0 +1,146 @@
+        LOGGER.info("Manager service: " + managerService);
+        LOGGER.info("Security service: " + securityService);
+
+        try {
+            final HttpServletRequest httpServletRequest =
+                    (HttpServletRequest) Executions.getCurrent().getNativeRequest();
+            LOGGER.info("httpServletRequest: {}", httpServletRequest);
+
+            final String appAuthCookieStr = JwtHelper.readCookie(httpServletRequest, "App_Auth");
+            final String signedAppAuthCookieStr =
+                    JwtHelper.readCookie(httpServletRequest, "Signed_App_Auth");
+            LOGGER.info("\n\n>>> appAuthCookieStr: {}", appAuthCookieStr);
+            LOGGER.info("\n\n>>> signedAppAuthCookieStr: {}", signedAppAuthCookieStr);
+
+            final UserType userType = JwtHelper.userFromJwt(managerService, appAuthCookieStr);
+
+            if (userType != null) {
+                UserSessionManager.setCurrentUser(userType);
+
+                final UserType currentUserType = UserSessionManager.getCurrentUser();
+                LOGGER.info("Current session logged-in currentUserType: " + currentUserType);
+                LOGGER.info("Current session logged-in username: " + currentUserType.getUsername());
+
+                final SecurityContext springSecurityContext = SecurityContextHolder.getContext();
+                LOGGER.info("springSecurityContext {}", springSecurityContext);
+
+                if (springSecurityContext != null) {
+                    final Authentication springAuthObj = springSecurityContext.getAuthentication();
+                    LOGGER.info("springAuthObj {}", springAuthObj);
+
+                    final User user =
+                            JwtHelper.convertFromUserType(currentUserType, securityService);
+                    LOGGER.info("user {}", user);
+
+                    final List<GrantedAuthority> authorities = getAuthorities(user.getRoles());
+
+                    final PreAuthenticatedAuthenticationToken preAuthenticatedAuthenticationToken =
+                            new PreAuthenticatedAuthenticationToken(user.getUsername(),
+                                    "", authorities);
+                    LOGGER.info("preAuthenticatedAuthenticationToken {}", preAuthenticatedAuthenticationToken);
+
+                    springSecurityContext.setAuthentication(preAuthenticatedAuthenticationToken);
+                } else {
+                    LOGGER.info("Spring SecurityContext is null!!");
+                }
+
+                LOGGER.info("Before redirect");
+                Executions.getCurrent().sendRedirect("/index.zul");
+                LOGGER.info("After redirect");
+            } else {
+                Executions.getCurrent().sendRedirect("/login.zul?error=2");
+            }
+        } catch (final Exception e) {
+            LOGGER.error("Exception in processing JWT/associating user {} - stackTrace {}", e.getMessage(),
+                    ExceptionUtils.getStackTrace(e));
```

Comment:
`LOGGER.error("Unable to process JWT/associate user", e);`

Reply:
Done.